packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/routers/router.dart

```diff
@@ -1,17 +1,232 @@
 {{> licence.dart }}
 
-import 'package:auto_route/auto_route.dart';
-
-import '../../feature_counter/views/counter_page.dart';
-import '../../feature_login/views/login_page.dart';
-import '../../feature_notifications/views/notifications_page.dart';
-
-@MaterialAutoRouter(
-  replaceInRouteName: 'Page,Route',
-  routes: <AutoRoute>[
-    MaterialRoute(page: CounterPage, initial: true),
-    MaterialRoute(page: LoginPage),
-    MaterialRoute(page: NotificationsPage),
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+
+import '../../feature_counter/di/counter_page_with_dependencies.dart';
+import '../../feature_enter_message/di/enter_message_with_dependencies.dart';
+import '../../feature_item_details/di/item_details_with_dependencies.dart';
+import '../../feature_items_list/di/items_list_with_dependencies.dart';
+import '../../feature_login/di/login_page_with_dependencies.dart';
+import '../../feature_notifications/di/notifications_page_with_dependencies.dart';
+import '../../feature_splash/di/splash_page_with_dependencies.dart';
+import '../common_blocs/coordinator_bloc.dart';
+import '../common_services/permissions_service.dart';
+import '../models/errors/error_model.dart';
+import '../models/item_model.dart';
+import 'go_router_refresh_stream.dart';
+import 'router_paths.dart';
+import 'router_permission_names.dart';
+
+part 'router.g.dart';
+
+class AppRouter {
+  AppRouter(this._context) {
```

Comment:
Why we inject the context here? Can't we inject the actual dependencies or if we need to defer it to use the Locator instead ?

Reply:
Done